src/background/uri-info.js

```diff
@@ -1,25 +1,72 @@
 import settings from './settings';
 
+const BLOCKLIST = [
+  { urlProperty: 'protocol', value: 'chrome:' },
+  { urlProperty: 'hostname', value: 'facebook.com' },
+  { urlProperty: 'hostname', value: 'www.facebook.com' },
+  { urlProperty: 'hostname', value: 'mail.google.com' },
+];
+
 /** encodeUriQuery encodes a string for use in a query parameter */
 function encodeUriQuery(val) {
   return encodeURIComponent(val).replace(/%20/g, '+');
 }
 
+/**
+ * Should we send a "badge" request to obtain the annotation count for the
+ * URL `uri`?
+ *
+ * @param {string} uri
+ * @return {boolean}
+ */
+function shouldQueryUri(uri) {
+  let url;
+
+  try {
+    url = new URL(uri);
+  } catch (e) {
+    return false;
+  }
+
+  // Make sure `uri` does not match ANY item in the blocklist
+  return BLOCKLIST.every(
```

Comment:
The current structure of the `BLOCKLIST` is optimized so that this test can short-circuit without a nested loop (`Array.prototoype.every` will short-circuit at the first false result).